tools/hidl/hidlcomp.cpp

```diff
@@ -4268,7 +4268,11 @@ void EspMessageInfo::write_esp()
     //method ==> copy
     outf("\nvoid C%s::copy(C%s &from)\n{\n", name_, name_);
     if (parent)
-        outf("\tC%s::copy(*dynamic_cast<C%s*>(&from));\n", parent, parent);
```

Comment:
minor:  I think this can be a static cast since C<name> will be derived from C<parent> if parent is non null.